src/app/+search-page/search-service/search.service.ts

```diff
@@ -373,6 +373,15 @@ export class SearchService implements OnDestroy {
     return '/search';
   }
 
+  /**
+   * Clear all request cache related to discovery objects
+   */
+  clearDiscoveryRequests() {
```

Comment:
I don't like this: you should never have to manually clear the cache. Instead, use requests with a responseMsToLive that is lower or 0. Then, in requestService, if a request for a URL is cached, a new request is made with a lower responseMsToLive value than the cached version, update that value in the cache before you check the validity. That should bypass the need for this method